torch/_inductor/codegen/cpp.py

```diff
@@ -1197,6 +1278,52 @@ def reduction(self, name, dtype, src_dtype, reduction_type, index, value):
             self.simd_vec = False
         return self.simd_vec
 
+    def is_supported_cmp(self, node: torch.fx.Node):
+        def get_node_dtype(node):
+            if type(node) == torch.fx.Node:
+                return None if "dtype" not in node.meta else node.meta["dtype"]
+            else:
+                return None
+
+        def get_cmp_dtypes(node: torch.fx.Node):
+            return get_node_dtype(node.args[-2]), get_node_dtype(node.args[-1])
+
+        assert len(node.args) >= 2
+        # cmp(x, y): y is a magic value like x >= 1
+        if type(node.args[-1]) in [int, float]:
+            return True
+        # cmp(x, y): x is a magic value like 1 >= y
+        if type(node.args[-2]) in [int, float]:
+            return False
+
+        left_dtype, right_dtype = get_cmp_dtypes(node)
+        if left_dtype is None or right_dtype is None:
+            # TODO(Eikan): Should be conservative?
```

Comment:
Shouldn't we start with being conservative and leave the aggressive option as TODO?

Reply:
Actually, it is a missing piece to record, deduce and propagate the data type of every expression. Hence, we could not get the real data type of the left expression or the right expression. I refined the comment. By the way, I'm working on the data type analysis.